samples/test-samples.cmake

```diff
@@ -38,6 +38,21 @@ else ()
   )
 endif ()
 
+# Read OS info if on Linux
+if (UNIX)
+  file(STRINGS "/etc/os-release" OS_INFO)
```

Comment:
You could consider using `REGEX` option to optimize this and avoid the foreach loop. Look for it in https://cmake.org/cmake/help/latest/command/file.html